src/vs/workbench/contrib/preferences/browser/settingsTree.ts

```diff
@@ -860,6 +948,72 @@ export class SettingArrayRenderer extends AbstractSettingRenderer implements ITr
 	}
 }
 
+export class SettingObjectRenderer extends AbstractSettingRenderer implements ITreeRenderer<SettingsTreeSettingElement, never, ISettingObjectItemTemplate> {
+	templateId = SETTINGS_OBJECT_TEMPLATE_ID;
+
+	renderTemplate(container: HTMLElement): ISettingObjectItemTemplate {
+		const common = this.renderCommonTemplate(null, container, 'list');
+
+		const objectWidget = this._instantiationService.createInstance(ObjectSettingWidget, common.controlElement);
+		objectWidget.domNode.classList.add(AbstractSettingRenderer.CONTROL_CLASS);
+		common.toDispose.add(objectWidget);
+
+		const template: ISettingObjectItemTemplate = {
+			...common,
+			objectWidget: objectWidget,
+		};
+
+		this.addSettingElementFocusHandler(template);
+
+		common.toDispose.add(objectWidget.onDidChangeList(e => this.onDidChangeMap(template, e)));
+
+		return template;
+	}
+
+	private onDidChangeMap(template: ISettingObjectItemTemplate, e: ISettingListChangeEvent<IObjectDataItem>): void {
+		if (template.context) {
+			const newValue = { ...template.context.scopeValue };
+
+			// first delete the existing entry, if present
+			if (e.originalItem.key.data) {
+				delete newValue[e.originalItem.key.data];
+			}
+
+			// then add the new or updated entry, if present
+			if (e.item?.key.data && e.item.value.data) {
+				newValue[e.item.key.data] = e.item.value.data;
+			}
+
+			function sortKeys<T extends object>(obj: T) {
```

Comment:
Not sure whether we want to sort the keys, the user might have ordered them in a certain way in their JSON, and this would wipe that out.